src/v/raft/group_manager.cc

```diff
@@ -203,4 +223,19 @@ void group_manager::setup_metrics() {
         sm::description("Number of raft groups"))});
 }
 
+ss::future<> group_manager::flush_groups() {
+    auto u = co_await _groups_mutex.get_units();
+
+    co_await ss::max_concurrent_for_each(
+      _groups.begin(),
+      _groups.end(),
+      32,
+      [this](const ss::lw_shared_ptr<consensus>& raft) {
+          if (_configuration.replica_max_not_flushed_bytes()) {
```

Comment:
nit: isn't it better to do this check outside the max_concurrent_for_each and avoid it entirely if max_bytes is not configured?

Reply:
since this is async situation may change between the scheduling points